metadata_service/proxy/neo4j_proxy.py

```diff
@@ -958,6 +965,61 @@ def get_user(self, *, id: str) -> Union[UserEntity, None]:
 
         return self._build_user_from_record(record=record, manager_name=manager_name)
 
+    def create_update_user(self, *, user: User) -> Tuple[User, bool]:
+        """
+        Create a user if it does not exist, otherwise update the user. Required
+        fields for creating / updating a user are validated upstream to this when
+        the User object is created.
+
+        :param user:
+        :return:
+        """
+        user_data = UserSchema().dump(user)
+        user_props = self._create_props_body(user_data, 'usr')
+
+        create_update_user_query = textwrap.dedent("""
+        MERGE (usr:User {key: $user_id})
+        on CREATE SET %s, usr.%s=timestamp()
+        on MATCH SET %s
+        RETURN usr, usr.%s = timestamp() as created
+        """ % (user_props, CREATED_EPOCH_MS, user_props, CREATED_EPOCH_MS))
+
+        try:
+            tx = self._driver.session().begin_transaction()
+            result = tx.run(create_update_user_query, user_data)
+
+            user_result = result.single()
+            if not user_result:
+                raise RuntimeError('Failed to create user with data %s' % user_data)
```

Comment:
I don't think this should be a RuntimeError since no result existing would likely be an issue with neo4j, not this program.

Reply:
I think that's a fair point. This specific error was used to keep in line with the existing paradigm for how Neo4j integration errors are raised, e.g.:

- https://github.com/amundsen-io/amundsenmetadatalibrary/blob/master/metadata_service/proxy/neo4j_proxy.py#L403
- https://github.com/amundsen-io/amundsenmetadatalibrary/blob/master/metadata_service/proxy/neo4j_proxy.py#L499

There are 6-7 places where this is used the the thought was that reusing this exception would make it easier to catch and handle these errors if we only need to watch for one type of exception. 

Given this, if you'd like to continue with a more descriptive exception hat are your thoughts on something akin to a `Neo4jErrorException` class and raise that here in place of the `RuntimeError` the code above (with migration of other runtime errors to this new exception occurring through a refactor in the future)?